src/sources/kafka.rs

```diff
@@ -120,121 +131,92 @@ impl SourceConfig for KafkaSourceConfig {
     }
 }
 
-fn kafka_source(
-    config: &KafkaSourceConfig,
+async fn kafka_source(
+    consumer: StreamConsumer,
+    key_field: String,
+    topic_key: String,
+    partition_key: String,
+    offset_key: String,
+    headers_key: String,
```

Comment:
These can probably be replaced by `&str` as we seem to borrow these fields before calling `log.insert`.

Reply:
I tried this, but ran into lifetime issues:

```
error[E0759]: `self` has lifetime `'life0` but it needs to satisfy a `'static` lifetime requirement
   --> src/sources/kafka.rs:110:21
    |
110 |       async fn build(&self, cx: SourceContext) -> crate::Result<super::Source> {
    |                      -^^^^
    |                      ||
    |                      |...is captured here...
    |                      this data with lifetime `'life0`...
...
113 | /         Ok(Box::pin(kafka_source(
114 | |             consumer,
115 | |             &self.key_field,
116 | |             &self.topic_key,
...   |
121 | |             cx.out,
122 | |         )))
    | |___________- ...and is required to live as long as `'static` here
    ```

Reply:
Yeah, since they are passed to an `async fn`, they would need to be `&'static` since the async has an unknown lifetime.